libqtile/layout/xmonad.py

```diff
@@ -267,6 +269,8 @@ def cmd_maximize(self):
 
     def configure(self, client, screen):
         "Position client based on order and sizes"
+        self.rect = screen
```

Comment:
Is `screen` an instance of `ScreenRect` here? Shouldn't we use `screen.get_rect()`?

Reply:
The existing naming is confusing.

`Layout.layout` takes a `screen`, which is really a ScreenRect.
https://github.com/qtile/qtile/blob/master/libqtile/layout/base.py#L54

It's called from `delegate_layout`, where it's clearer that it's passing a ScreenRect.
https://github.com/qtile/qtile/blob/master/libqtile/layout/base.py#L288

It would be nice to rename the `screen` parameter of `configure` and `show` to be called `rect` or `screen_rect`, but that would break the API.

Reply:
Let's break the API if some other name makes much more sense and avoids confusion. Or maybe we should break the API on a type perspective and give a Screen instead of a ScreenRect?

Reply:
I would lean towards only passing a ScreenRect, since that's all the receiver needs.

Should I rename it in this PR, or as a follow-up?

Reply:
This touches a lot of things. I'd like to make it a separate PR if that's OK with you?

Reply:
Looks like there was one more place we need to pass the `rect` down: `TreeTab._create_drawer`. Doing that in this PR too...

Reply:
I made the rename a separate PR. I don't think I can base it on my other branch/PR, but here's a diff until we merge the first PR: https://github.com/mikelward/qtile/pull/1/commits/a93e070aa226452784fc468d753c62fde5cd40b2

Reply:
https://github.com/qtile/qtile/pull/1838 for renaming `screen` to `screen_rect`.

Reply:
Awesome, thanks!